cmd/soroban-rpc/lib/preflight/src/lib.rs

```diff
@@ -196,11 +198,56 @@ fn preflight_host_function_or_maybe_panic(
         error: null_mut(),
         result: result_cstr.into_raw(),
         footprint: fp_cstr.into_raw(),
+        // TODO
+        auth: null_mut(),
         cpu_instructions: budget.get_cpu_insns_count(),
         memory_bytes: budget.get_mem_bytes_count(),
     })
 }
 
+fn recoded_auth_payloads_to_c(
+    payloads: Vec<RecordedAuthPayload>,
+) -> Result<*mut *mut libc::c_char, Box<dyn error::Error>> {
+    // Get a vector of base64-encoded ContractAuths
+    let mut out_vec: Vec<*mut libc::c_char> = Vec::new();
+    for p in payloads.iter() {
+        let c_str = CString::new(recoded_auth_payload_to_xdr(p).to_xdr_base64()?)?.into_raw();
+        out_vec.push(c_str);
+    }
+
+    // Add the ending NULL
+    out_vec.push(null_mut());
+
+    // Make sure length and capacity are the same
+    // (this allows using the length as the capacity when deallocating the vector)
+    out_vec.shrink_to_fit();
+    assert!(out_vec.len() == out_vec.capacity());
+
+    // Get the pointer to our vector, we will deallocate it in free_preflight_result()
+    // TODO: replace by `out_vec.into_raw_parts()` once the API stabilizes
+    let ptr = out_vec.as_mut_ptr();
+    mem::forget(out_vec);
+
+    Ok(ptr)
+}
+
+fn recoded_auth_payload_to_xdr(payload: &RecordedAuthPayload) -> ContractAuth {
+    let address_with_nonce = match (payload.address.clone(), payload.nonce) {
+        (Some(a), Some(n)) => Some(AddressWithNonce {
+            address: a,
+            nonce: n,
+        }),
+        // TODO: can the address and the nonce really be present independently?
+        _ => None,
+    };
+    ContractAuth {
+        address_with_nonce: address_with_nonce,
+        root_invocation: payload.invocation.clone(),
+        // TODO: ???
+        signature_args: Default::default(),
```

Comment:
I am not sure how to fill this info.

Any ideas @paulbellamy ?

Reply:
Good question. I'd expect the args passed to `auth_required`, but those should already be captured as part of the `root_invocation`, so I'm not sure what this is. @dmkozh ?

Reply:
From @dmkozh We should leave this blank (or return a different type to exclude it). This is for the client to fill in when submitting the txn. Updated the comment.